src/Fission.bs.js

```diff
@@ -4,109 +4,191 @@
 var Axios = require("bs-axios/src/axios.js");
 var Axios$1 = require("axios");
 
-var baseURL = "http://localhost:1337";
-
-var env_username = "ca2c70bc13298c5109ee";
-
-var env_password = "VlBgonAFjZon2wd2VkTR3uc*p-XMd(L_Zf$nFvACpHQShqJ_Hp2Pa";
+function $$await(promise) {
+  return promise.then((function (response) {
+                  return Promise.resolve(response.data);
+                })).catch((function (prim) {
+                return Promise.resolve(prim);
+              }));
+}
 
 var octetHeader = Axios.$$Headers[/* fromObj */0]({
       "content-type": "application/octet-stream"
     });
 
-function convAuth(auth) {
-  return {
-          username: auth[/* username */0],
-          password: auth[/* password */1]
-        };
+function Simple(Base) {
+  var ipfsURL = Base[/* baseUrl */0] + "/ipfs";
```

Comment:
Looks up by index of an array, which means that JS/TS consumers would call it like this:

```JS
const fsn = Fission.Simple(["https://hostless.dev/"]);
const myURL = fsn[2].("Qm....xyz")
```

Not idea. It looks like it's possible to do _actual_ classes with some annotations, but I don't love the hints all over the place https://bucklescript.github.io/docs/en/class